pkg/scanner/job-manager.go

```diff
@@ -179,22 +180,17 @@ func (s *Scanner) deleteJob(job *batchv1.Job) {
 
 // Due to K8s names constraint we will take the image name w/o the tag and repo
 func getSimpleImageName(imageName string) string {
-	repoEnd := strings.LastIndex(imageName, "/")
-	imageName = imageName[repoEnd+1:]
-
-	digestStart := strings.LastIndex(imageName, "@")
-	// remove digest if exists
-	if digestStart != -1 {
-		return imageName[:digestStart]
+	ref, err := reference.ParseNormalizedNamed(imageName)
+	if err != nil {
+		log.Errorf("failed to parse image name. name=%v: %v", imageName, err)
+		return imageName
 	}
 
-	tagStart := strings.LastIndex(imageName, ":")
-	// remove tag if exists
-	if tagStart != -1 {
-		return imageName[:tagStart]
-	}
+	refName := ref.Name()
+	// Take only image name from repo path (ex. solsson/kafka ==> kafka)
+	repoEnd := strings.LastIndex(refName, "/")
 
-	return imageName
+	return refName[repoEnd+1:]
```

Comment:
ParseNormalizedNamed should guard us, but maybe worth checking that we don't crash on imageName that ends with /

Reply:
ParseNormalizedNamed will failed to parse when it's an invalid reference format.
I've change the code to return error when ParseNormalizedNamed() fails, if we can't normalized it the job creation will probably failed so better to fail earlier.